### PR TITLE
[WIP] Prune internal client and internal types from printer

### PR DESCRIPTION
**Pull request by yue9944882**
this pull introduces following changes into kubectl desdescriber:

1. Prunes internal clientset dependency
2. Switch all internal types to versioned types

/sig cli
/sig api-machinery
/kind cleanup
/priority important-soon

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/history.go [hunk 1]
```diff
@@ -168,12 +166,8 @@ func (h *DeploymentHistoryViewer) ViewHistory(namespace, name string, revision i
 
 func printTemplate(template *corev1.PodTemplateSpec) (string, error) {
 	buf := bytes.NewBuffer([]byte{})
-	internalTemplate := &api.PodTemplateSpec{}
-	if err := apiv1.Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(template, internalTemplate, nil); err != nil {
```
Comment (yue9944882) [inline pkg/kubectl/history.go]: this change makes me happy 🙂

Comment (adohe) [reply pkg/kubectl/history.go]: 👍 👍

**Pull request closed without merging**